From 40eb9ce75518817762a0eac4a93016ab817add89 Mon Sep 17 00:00:00 2001
From: Lee Duncan <lduncan@suse.com>
Date: Sat, 1 Feb 2020 10:23:04 -0800
Subject: [PATCH] Fix 586 compile issue and remove -Werror

Using -Werror causes any issue to break the build, whereas
I'd rather let the build continue and address the issue.

Also, fixed one signed-vs-unsigned compare for time_t, which
shows up only on 586 (32-bit).
---
 configure.ac | 2 +-
 isnsdd.c     | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index e4f3995..d956e58 100644
--- a/configure.ac
+++ b/configure.ac
@@ -17,7 +17,7 @@ AC_PATH_PROG(SH, sh)
 dnl C Compiler features
 AC_C_INLINE
 if test "$GCC" = "yes"; then
-        CFLAGS="-Wall -Werror -Wextra $CFLAGS"
+        CFLAGS="-Wall -Wextra $CFLAGS"
 	CPPFLAGS="$CPPFLAGS -D_GNU_SOURCE"
 fi
 
diff --git a/isnsdd.c b/isnsdd.c
index 58825cc..9cedb9f 100644
--- a/isnsdd.c
+++ b/isnsdd.c
@@ -401,7 +401,7 @@ check_portal_registration(__attribute__((unused))void *ptr)
 			continue;
 
 		last_modified = isns_object_last_modified(obj);
-		if (last_modified + 2 * interval > now) {
+		if ((time_t)(last_modified + 2 * interval) > now) {
 			good_portals++;
 			continue;
 		}
